lib/sql.js

```diff
@@ -1508,3 +1508,274 @@ SQLConnector.prototype.getInsertedId = function(model, info) {
+  if (!cb && typeof options === 'function') {
+    cb = options;
+    options = {};
+  }
+  options = options || {};
+
+  var self = this;
+  var calls = [function(callback) {
+    self.execute(self.buildQueryTables(options), callback);
+  }];
+
+  if (options.views) {
+    calls.push(function(callback) {
+      self.execute(self.buildQueryViews(options), callback);
+    });
+  }
+  async.parallel(calls, function(err, data) {
+    if (err) {
+      cb(err, data);
+    } else {
+      var merged = [];
+      merged = merged.concat(data.shift());
+      if (data.length) {
+        merged = merged.concat(data.shift());
+      }
+      cb(err, merged);
+    }
+  });
+};
+// Due to the different implementation structure of information_schema across
+// connectors, each connector will have to generate its own query
+SQLConnector.prototype.buildQueryColumns = function(schema, table) {
+  throw new Error(g.f('{{buildQueryColumns}} must be implemented by the connector'));
+};
+
+SQLConnector.prototype.buildPropertyType = function(columnDefinition, options) {
+  throw new Error(g.f('{{buildPropertyType}} must be implemented by the connector'));
+};
+
+/*!
+ * Normalize the arguments
+ * @param table string, required
+ * @param options object, optional
+ * @param cb function, optional
+ */
+SQLConnector.prototype.getArgs = function(table, options, cb) {
+  throw new Error(g.f('{{getArgs}} must be implemented by the connector'));
+};
+
+/**
+ * Discover model properties from a table
+ * @param {String} table The table name
+ * @param {Object} options The options for discovery
+ * @param {Function} [cb] The callback function
+ *
+ */
+SQLConnector.prototype.discoverModelProperties = function(table, options, cb) {
+  var self = this;
+  var args = self.getArgs(table, options, cb);
+  var schema = args.schema;
+  var connectorName = self.dataSource.connector.name;
+
+  table = args.table;
+  options = args.options;
+
+  if (typeof(self.getDefaultSchema) === 'function' && !schema) {
+    schema = self.getDefaultSchema();
+  }
+
+
+  // Recommended MySQL 5.7 Boolean scheme. See
+  // http://dev.mysql.com/doc/refman/5.7/en/numeric-type-overview.html
+  // Currently default is the inverse of the recommendation for backward compatibility.
+  var setDefaultOptions = ['mysql'];
+  if (setDefaultOptions.indexOf(connectorName) > -1) {
+    var defaultOptions = {
+      treatCHAR1AsString: false,
+      treatBIT1AsBit: true,
+      treatTINYINT1AsTinyInt: true,
+    };
+
+    for (var opt in defaultOptions) {
+      if (defaultOptions.hasOwnProperty(opt) && !options.hasOwnProperty(opt)) {
+        options[opt] = defaultOptions[opt];
+      }
+    }
+  }
+  cb = args.cb;
+
+  var sql = self.buildQueryColumns(schema, table);
+  var callback = function(err, results) {
+    if (err) {
+      cb(err, results);
+    } else {
+      results.map(function(r) {
+        r.type = self.buildPropertyType(r, options);
+        var setNullableConnectors = ['mysql', 'db2', 'dashdb'];
```

Comment:
@raymondfeng Sounds reasonable, we will add function `prototype.setDefaultOptions()` in mysql and do same handling here: add `prototype.setNullableConnectors` for mysql, db2, dashdb.